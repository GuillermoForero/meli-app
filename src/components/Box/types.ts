export interface BoxProps {
  width?: string;
  height?: string;
  minHeight?: string;
  justifyContent?: string;
  alignItems?: string;
  alignSelf?: string;
  justifySelf?: string;
  flexDirection?: string;
  padding?: string;
  margin?: string;
}
