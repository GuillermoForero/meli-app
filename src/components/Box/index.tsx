import styled from 'styled-components';

import { BoxProps } from './types';
//I created this reusable component because for SEO it would not affect much, due to the behavior of the website itself and the react
const Box = styled.div<BoxProps>`
  min-height: ${({ minHeight }) => (minHeight ? minHeight : false)};
  display: flex;
  justify-content: ${({ justifyContent }) => justifyContent || 'flex-start'};
  align-items: ${({ alignItems }) => alignItems || 'flex-start'};
  align-self: ${({ alignSelf }) => alignSelf || false};
  flex-direction: ${({ flexDirection }) => flexDirection || 'row'};
  box-sizing: border-box;
  height: ${({ height }) => height || 'auto'};
  width: ${({ width }) => width || '100%'};
  padding: ${({ padding }) => padding || '0'};
  margin: ${({ margin }) => margin || '0'};
`;

export default Box;
