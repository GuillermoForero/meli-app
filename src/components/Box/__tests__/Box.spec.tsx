import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Box from '../';

describe('[Box]', () => {
  const cases = [
    {
      name: 'default props',
    },
    {
      name: 'custom size props',
      size: {
        sm: '40px',
        xs: '80px',
        md: '100px',
        lg: '120px',
        xlg: '150px',
      },
    },
  ];

  cases.forEach(c => {
    it(`Renders a Box Component with ${c.name}`, () => {
      const { container } = render(<Box width={c.size} height={c.size} />);

      expect(container).toMatchSnapshot();
    });
  });
});
