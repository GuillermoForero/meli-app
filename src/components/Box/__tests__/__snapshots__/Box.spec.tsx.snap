// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Box] Renders a Box Component with custom size props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  margin: 0px;
  padding: 0px;
}

@media (min-width:576px) and (max-width:767.98px) {
  .c0 {
    width: 40px;
  }
}

@media (max-width:575.98px) {
  .c0 {
    width: 80px;
  }
}

@media (min-width:768px) and (max-width:991.98px) {
  .c0 {
    width: 100px;
  }
}

@media (min-width:992px) and (max-width:1199.98px) {
  .c0 {
    width: 120px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 150px;
  }
}

@media (min-width:576px) and (max-width:767.98px) {
  .c0 {
    height: 40px;
  }
}

@media (max-width:575.98px) {
  .c0 {
    height: 80px;
  }
}

@media (min-width:768px) and (max-width:991.98px) {
  .c0 {
    height: 100px;
  }
}

@media (min-width:992px) and (max-width:1199.98px) {
  .c0 {
    height: 120px;
  }
}

@media (min-width:1200px) {
  .c0 {
    height: 150px;
  }
}

<div>
  <div
    class="c0"
    height="[object Object]"
    width="[object Object]"
  />
</div>
`;

exports[`[Box] Renders a Box Component with default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  margin: 0px;
  padding: 0px;
  width: 100%;
  height: auto;
}

<div>
  <div
    class="c0"
  />
</div>
`;
